src/modules/dashboard/components/Extensions/ExtensionDetails.tsx

```diff
@@ -0,0 +1,342 @@
+              )}
+            />
+            <Route
+              exact
+              path={COLONY_EXTENSION_SETUP_ROUTE}
+              component={() => {
+                if (
+                  !canInstall ||
+                  (installedExtension?.details.initialized &&
+                    !installedExtension?.details.missingPermissions.length)
+                ) {
+                  return <Redirect to={extensionUrl} />;
+                }
+                return installedExtension ? (
+                  <ExtensionSetup
+                    extension={extension}
+                    installedExtension={installedExtension}
+                    colonyAddress={colonyAddress}
+                  />
+                ) : (
+                  <Redirect to={extensionUrl} />
+                );
+              }}
+            />
+          </Switch>
+        </div>
+      </div>
+      <aside>
+        <div className={styles.extensionDetails}>
+          <hr className={styles.headerLine} />
+          <div className={styles.buttonWrapper}>
+            {!onSetupRoute &&
+              canInstall &&
+              (!installedExtension?.details.initialized ||
+                !!installedExtension?.details.missingPermissions.length) && (
+                <ExtensionActionButton
+                  colonyAddress={colonyAddress}
+                  installedExtension={installedExtension}
+                  extension={extension}
+                />
+              )}
+          </div>
+          <Table appearance={{ theme: 'lined' }}>
+            <TableBody>
+              {tableData.map(({ label, value }) => (
+                <TableRow key={label.id}>
+                  <TableCell className={styles.cellLabel}>
+                    <FormattedMessage {...label} />
+                  </TableCell>
+                  <TableCell className={styles.cellData}>{value}</TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+          {canInstall &&
+          extension.uninstallable &&
+          installedExtension &&
+          !installedExtension.details.deprecated ? (
+            <div className={styles.buttonUninstall}>
+              <DialogActionButton
+                dialog={ConfirmDialog}
+                dialogProps={{
+                  heading: MSG.headingDeprecate,
+                  children: <FormattedMessage {...MSG.textDeprecate} />,
+                }}
+                appearance={{ theme: 'blue' }}
+                submit={ActionTypes.COLONY_EXTENSION_DEPRECATE}
+                error={ActionTypes.COLONY_EXTENSION_DEPRECATE_ERROR}
+                success={ActionTypes.COLONY_EXTENSION_DEPRECATE_SUCCESS}
+                text={MSG.buttonDeprecate}
+                values={{ colonyAddress, extensionId }}
+              />
+            </div>
+          ) : null}
+          {canInstall &&
+          extension.uninstallable &&
+          installedExtension?.details.deprecated ? (
```

Comment:
I would move those 3 huge checks from each part of this component at the top and name it - it would be better for reading/understanding. But of course yours call